builtin/game/respawn.lua

```diff
@@ -0,0 +1,39 @@
+-- Minetest: builtin/game/respawn.lua
+
+function core.show_death_formspec(player)
+	core.show_formspec(player:get_player_name(), "builtin:death", [[
```

Comment:
This allows any player to respawn instantly by sending this formspec to the server. 
The code in `register_on_player_receive_fields` may only be executed when the player died but did not respawn yet.

Reply:
Also: This is just a regular formspec. Delayed `show_formspec` calls may override this one; so the player cannot respawn.

What if the player joined with 0 HP? They're dead. Show the respawn dialogue to them.

Reply:
Nice, I forgot to test an important thing. Will investigate.

Reply:
Fixed the first comment.
> What if the player joined with 0 HP?

The server sends the HP when a player joins, so on_dieplayer will be triggered (tested).

Reply:
> This is just a regular formspec.

I thought all formspecs were the same. What's a non-regular formspec?

> Delayed `show_formspec` calls may override this one; so the player cannot respawn.

I did some testing and it seems the current death formspec (both the C++ and CSM versions) has the same issue.